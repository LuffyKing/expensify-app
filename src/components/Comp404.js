import React from 'react';
import {Link} from 'react-router-dom';
const Comp404 = (props) => {
  return (
    <div>404!
      <Link to="/">Go Home</Link>
    </div>
  )
}

export default Comp404;
