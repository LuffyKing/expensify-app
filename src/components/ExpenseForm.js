import React from 'react';
import moment from 'moment';
import {SingleDatePicker} from 'react-dates';
import 'react-dates/initialize';
const now = moment();
class ExpenseForm extends React.Component{
  constructor(props){
    super(props);
    this.state = {
      description: props.expense ? props.expense.description:'',
      note: props.expense ? props.expense.note:'',
      amount: props.expense ? props.expense.amount:'',
      createdAt: props.expense ? moment(props.expense.createdAt):moment(),
      calendarFocused:false,
      error:''

    };
  }

  onDescriptionChange = (e) => {
    const description = e.target.value;
    this.setState(() => ({description}));

  };
  onTextAreaChange = (e) => {
    const note = e.target.value;
    this.setState(()=>({note}));

  };
  onAmountChange = (e) => {
    const amount = e.target.value;
    const numTest = new RegExp(/^[1-9]\d*\.?\d{0,2}$/, 'i');
    if(numTest.test(amount)||!amount){
      this.setState(() => ({amount}));
    };


  };
  onDateChange = (createdAt) => {
    if(createdAt){
      this.setState(()=>({createdAt}));
    }
  };
  onFocusChange= ({focused}) =>{
    this.setState(() => ({calendarFocused: focused}));
  };
  onSubmit = (e) => {
    e.preventDefault();
    if(!this.state.description||!this.state.amount){
      const error='Please provide description and amount!';
      this.setState(() =>({error}));
    }
    else{
      const error='';
      this.setState(() =>({error}));
      this.props.onSubmit({
        description: this.state.description,
        amount: +this.state.amount,
        note:this.state.note,
        createdAt:this.state.createdAt.valueOf(),
        note:this.state.note,
      })
    }

  };
  render(){
    return(

        <form onSubmit={this.onSubmit} className='form'>
          {

            this.state.error&& <h3>Error!! {this.state.error}</h3>
          }
          <input
            type="text"
            autoFocus
            placeholder="Description"
            value={this.state.description}
            onChange={this.onDescriptionChange}
            className="text-input"
          />

          <input
            type="text"
            className="text-input"
            value={this.state.amount}
            onChange={this.onAmountChange}
            placeholder="Amount"/>
          <SingleDatePicker
            date={this.state.createdAt}
            onDateChange={this.onDateChange}
            focused={this.state.calendarFocused}
            onFocusChange={this.onFocusChange}
            numberOfMonths={1}
            isOutsideRange={() => false}
            >
          </SingleDatePicker>
          <textarea
            placeholder="Add a note for your expense (optional)"
            value = {this.state.note}
            onChange={this.onTextAreaChange}
            className="textarea"
            >

          </textarea>
          <div><button className="genButton">Save Expense</button></div>
        </form>
    );
  };
}
export default ExpenseForm;
