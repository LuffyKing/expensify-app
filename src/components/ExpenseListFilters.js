import React from 'react';
import {connect} from 'react-redux';
import {setTextFilter, sortByDate, sortByAmount, setEndDate, setStartDate} from '../actions/filters';
import {Link} from 'react-router-dom';
import {DateRangePicker} from 'react-dates';
import 'react-dates/initialize';

export class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null
  };
  onDatesChange = ({startDate, endDate}) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };
  onFocusChange = (calendarFocused) => {
    this.setState(() => ({calendarFocused}));
  };
  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  };
  onSortChange = (e) => {

    switch (e.target.value) {
      case 'date':
        {
          this.props.sortByDate();
          break;

        }

      case 'amount':
        {
          this.props.sortByAmount();
          break;
        }
      default:
        {
          break;
        }

    }
 
  };

  render() {
    return (<div className="content-container">
      <div className="input-group">
        <div className="input-group__item">
          <input placeholder="Search Expenses" type="text" className="text-input" value={this.props.filters.text} onChange={this.onTextChange}/>
        </div>
        <div className="input-group__item">
          <select className="select" value={this.props.filters.sortBy} onChange={this.onSortChange}>
            <option value="date">Date</option>
            <option value="amount">Amount</option>
          </select>
        </div>
        <div className="input-group__item">
          <DateRangePicker startDate={this.props.filters.startDate} endDate={this.props.filters.endDate} onDatesChange={this.onDatesChange} focusedInput={this.state.calendarFocused} onFocusChange={this.onFocusChange} startDateId="start_date" endDateId="end_date" numberOfMonths={1} showClearDates={true} isOutsideRange={() => false}></DateRangePicker>
        </div>
      </div>

    </div>);

  };
}

// setup value and onChange for select

const mapDispatchToProps = (dispatch) => {
  return {
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate)),
    setTextFilter: (value) => dispatch(setTextFilter(value)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount())
  };
};
const mapStateToProps = (state) => {
  return {filters: state.filters};
};
export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);
