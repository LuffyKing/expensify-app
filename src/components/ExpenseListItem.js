import React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';
export const ExpenseListItem = ({index,description,amount,createdAt,id}) => {
  return (

  <div>
    <h3><Link to={`/edit/${id}`}>{description}</Link></h3>
    <p>
      {numeral(amount).format('$0,0.00')}
      -
      {moment(createdAt).format('Do MMMM, YYYY')}
    </p>
  </div>
    )
};


export default connect()(ExpenseListItem);
