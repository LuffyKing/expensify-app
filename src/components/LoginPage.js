import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from '../actions/auth';
/*
<input type="text" placeholder="Username" id="username"/>
<input type="text" placeholder="Password" id="password"/>
*/
export const LoginPage = ({startLogin}) =>{
  return(
    <div>
      <h1>Login to Expensify</h1>
      <button onClick={startLogin}>Login</button>
    </div>
  )
};
const mapDispatchToProps = (dispatch) => ({startLogin:()=> dispatch(startLogin())});
export default connect(undefined, mapDispatchToProps)(LoginPage);
