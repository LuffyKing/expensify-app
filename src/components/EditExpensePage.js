import React from 'react';
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {editExpense, removeExpense} from '../actions/expenses';

export class EditExpensePage extends React.Component{
  onRemove= () => {

    this.props.removeExpense(this.props.expense.id);
    this.props.history.push('/');
  }
  onSubmit = (expense) => {
    this.props.editExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  }
  render(){
    return(
      <div>

        <ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit}></ExpenseForm>
        <button
          onClick={this.onRemove}
          >
        Remove
      </button>

      </div>

    );
  }


}

// const EditExpensePage = (props) => {
//   return (
    // <div>
    //   Editing expense with id of {props.match.params.id}
    //   <ExpenseForm expense={props.expense} onSubmit={(expense)=>{
    //     props.dispatch(editExpense(props.match.params.id, expense));
    //     props.history.push('/');
    //   }}></ExpenseForm>
    //   <button
    //     onClick={() => {
    //       console.log(props);
    //       let id = props.match.params.id;
    //       props.dispatch(removeExpense({id}));
    //       props.history.push('/');
    //    }
    // }>
    //   Remove
    // </button>
    //
    // </div>
//   )
// }

const mapDispatchToProps = (dispatch)=>{
  return {
    removeExpense:(id)=>{return dispatch(removeExpense({id}))},
    editExpense:(id, expense)=>dispatch(editExpense(id, expense))
  };
};

const mapStateToProps = (state, props) => {
  return{
    expense:state.expenses.find((expense) => {
          return expense.id === props.match.params.id;
    })
  };
};
export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
