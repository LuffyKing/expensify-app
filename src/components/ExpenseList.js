import React from 'react';
import {connect} from 'react-redux';
import ExpenseListItem from '../components/ExpenseListItem';
import selectExpenses from '../selector/expenses';
export const ExpenseList = (props) =>{
  return(
    <div>
      <div className="list-header">
        <div className="show-for-mobile">Expenses</div>
        <div className="show-for-desktop">Expense</div>
        <div className="show-for-desktop">Amount</div>
      </div>
      <div className="list-body">
        {
          props.expenses.length === 0 ? (
            <div className="list-item list-item--message">No Expenses</div>
          ): props.expenses.map((expense, index)=>{
            index+=1;

            return(

              <ExpenseListItem {...expense} index={index} key={expense.id}></ExpenseListItem>
            )
          })
        }
      </div>

    </div>
  );
};

const mapStateToProps  = (state) => {
  return {
     expenses: selectExpenses(state.expenses, state.filters)
  };
};


export default connect(mapStateToProps)(ExpenseList);
