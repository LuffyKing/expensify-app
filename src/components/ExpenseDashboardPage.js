import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpenseSummary from './ExpensesSummary';
const ExpenseDashboardPage = () => (
  <div>
    <ExpenseSummary></ExpenseSummary>
    <ExpenseListFilters></ExpenseListFilters>
    <ExpenseList></ExpenseList>
  </div>
);
export default ExpenseDashboardPage;
