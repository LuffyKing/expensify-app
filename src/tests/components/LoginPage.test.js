import React from 'react';
import {shallow} from 'enzyme';
import {LoginPage} from '../../components/LoginPage';

let wrapper, startLogin;
beforeEach(() => {
  startLogin = jest.fn();
  wrapper = shallow(<LoginPage startLogin={startLogin}></LoginPage>);

});

test('should render login page',() => {
  expect(wrapper).toMatchSnapshot();
});
test('should call startLogin',() => {
  wrapper.find('button').simulate('click');
  expect(startLogin).toHaveBeenCalled();
});
