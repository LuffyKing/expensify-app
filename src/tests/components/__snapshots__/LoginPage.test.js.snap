// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render login page 1`] = `
<div>
  <h1>
    Login to Expensify
  </h1>
  <button
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
